lib/rubocop/cop/rails/pluck_in_where.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # This cop identifies places where `pluck` is used in `where` query methods
+      # and can be replaced with `select`.
+      #
+      # Since `pluck` is an eager method and hits the database immediately,
+      # using `select` helps to avoid additional database queries.
+      #
+      # @example
+      #   # bad
+      #   Post.where(user_id: User.active.pluck(:id))
+      #
+      #   # good
+      #   Post.where(user_id: User.active.select(:id))
+      #
+      class PluckInWhere < Cop
+        MSG = 'Use `select` instead of `pluck` within `where` query method.'
+        WHERE_METHODS = %i[where rewhere].freeze
+
+        def_node_matcher :pluck_call?, <<~PATTERN
+          (send _ :pluck ...)
+        PATTERN
+
+        def on_send(node)
+          add_offense(node, location: :selector) if pluck_call?(node) && in_where?(node)
+        end
+
+        def autocorrect(node)
+          lambda do |corrector|
+            corrector.replace(node.loc.selector, 'select')
+          end
+        end
+
+        private
+
+        def in_where?(node)
+          send_node = node.each_ancestor(:send).first
+          WHERE_METHODS.include?(send_node&.method_name)
```

Comment:
Why are safe navigation operator (`&.`) used?

Reply:
`pluck` can be not within other send node, so `send_node` can be nil. 
I rewrote this line without `&` to make it a little bit clear.